jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1896,6 +1896,20 @@ static void loadLibraryWithPath(String libName, ClassLoader loader, String libra
 		}
 	}
 	byte[] message = ClassLoader.loadLibraryWithPath(com.ibm.oti.util.Util.getBytes(libName), loader, libraryPath == null ? null : com.ibm.oti.util.Util.getBytes(libraryPath));
+
+	if (System.internalGetProperties().getProperty("os.name").equals("Mac OS X")) { //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
Please add a comment describing why this code is added and is OSX specific

Reply:
fixed